File: wagerproof-ios-native/Wagerproof/Features/Agents/Components/CollapsingAgentsHeader.swift
import SwiftUI
import WagerproofDesign
import WagerproofModels

// The Agents hub's pixel office ("Agent HQ") used to be a scroll-driven floating
// widget that collapsed into a draggable corner minimap. That was replaced with
// a plain scrolling hero row (see `AgentsView.officeRow`) so it scrolls away
// naturally with the rest of the list. The only surviving piece is the agency
// stats pill below, which now rides in the office hero's top-trailing corner.

/// Compact agency stats rendered as a liquid-glass pill that floats over the
/// office hero. Mirrors `CompanyDashboardBanner`'s computations.
struct AgencyStatsPill: View {
    let agents: [AgentWithPerformance]

    private var totalNetUnits: Double {
        agents.reduce(0) { $0 + ($1.performance?.netUnits ?? 0) }
    }
    private var winRateAverage: Double {
        let withPerf = agents.compactMap { $0.performance }.filter { $0.totalPicks > 0 }
        guard !withPerf.isEmpty else { return 0 }
        let total = withPerf.reduce(0.0) { acc, perf in
            let settled = perf.wins + perf.losses
            guard settled > 0 else { return acc }
            return acc + Double(perf.wins) / Double(settled)
        }
        return total / Double(withPerf.count)
    }
    private var activeCount: Int { agents.filter { $0.agent.isActive }.count }
    private var unitsLabel: String {
        let sign = totalNetUnits >= 0 ? "+" : ""
        return String(format: "%@%.2fu", sign, totalNetUnits)
    }

    var body: some View {
        HStack(spacing: 5) {
            Text(unitsLabel)
                .foregroundStyle(totalNetUnits >= 0 ? Color(hex: 0x4ADE80) : Color(hex: 0xF87171))
            Text("·").foregroundStyle(.white.opacity(0.5))
            Text(String(format: "%.0f%%", winRateAverage * 100))
                .foregroundStyle(.white)
            Text("·").foregroundStyle(.white.opacity(0.5))
            Text("\(activeCount)/\(agents.count)")
                .foregroundStyle(.white)
        }
        .font(.system(size: 11, weight: .heavy, design: .rounded))
        .lineLimit(1)
        .fixedSize()
        .padding(.horizontal, 11)
        .padding(.vertical, 6)
        // Dark backing + white text — the pill sits over the colorful office.
        .background(Color.black.opacity(0.45), in: Capsule())
        .overlay(Capsule().strokeBorder(Color.white.opacity(0.18), lineWidth: 0.5))
    }
}

File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/agents/components/CollapsingAgentsHeader.kt
package com.wagerproof.app.features.agents.components

import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.wagerproof.core.design.components.liquidGlassBackground
import com.wagerproof.core.design.tokens.AppColors
import com.wagerproof.core.models.AgentWithPerformance
import java.util.Locale

/**
 * Compact agency stats pill that floats over the office hero's top-trailing
 * corner — the only surviving piece of iOS `CollapsingAgentsHeader.swift`.
 * Net units (P/L-tinted) · avg win% · active/total, on a liquid-glass capsule.
 */
@Composable
fun AgencyStatsPill(
    agents: List<AgentWithPerformance>,
    modifier: Modifier = Modifier,
) {
    val totalNetUnits = agents.sumOf { it.performance?.netUnits ?: 0.0 }
    val winRateAverage = avgWinRate(agents)
    val activeCount = agents.count { it.agent.isActive }

    Row(
        modifier
            .liquidGlassBackground(CircleShape)
            .padding(horizontal = 11.dp, vertical = 6.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        val dot: @Composable (String, Color) -> Unit = { text, color ->
            Text(
                text,
                color = color,
                fontSize = 11.sp,
                fontWeight = FontWeight.Black,
                fontFamily = FontFamily.Default,
                maxLines = 1,
                modifier = Modifier.padding(horizontal = 2.5.dp),
            )
        }
        dot(netUnitsLabel(totalNetUnits), if (totalNetUnits >= 0) AppColors.statsPillProfit else AppColors.statsPillLoss)
        dot("·", Color.White.copy(alpha = 0.5f))
        dot(String.format(Locale.US, "%.0f%%", winRateAverage * 100), Color.White)
        dot("·", Color.White.copy(alpha = 0.5f))
        dot("$activeCount/${agents.size}", Color.White)
    }
}
